_episodes/06-exploring_history.md

```diff
@@ -12,6 +12,605 @@ objectives:
+
+:::::::::::::::::::::::::::::::::::::::::  callout
+
+## Simplifying the Common Case
+
+If you read the output of `git status` carefully,
+you'll see that it includes this hint:
+
+```output
+(use "git checkout -- <file>..." to discard changes in working directory)
+```
+
+As it says,
+`git checkout` without a version identifier restores files to the state saved in `HEAD`.
+The double dash `--` is needed to separate the names of the files being recovered
+from the command itself:
+without it,
+Git would try to use the name of the file as the commit identifier.
+
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+The fact that files can be reverted one by one
+tends to change the way people organize their work.
+If everything is in one large document,
+it's hard (but not impossible) to undo changes to the introduction
+without also undoing changes made later to the conclusion.
+If the introduction and conclusion are stored in separate files,
+on the other hand,
+moving backward and forward in time becomes much easier.
+
+:::::::::::::::::::::::::::::::::::::::  challenge
+
+## Recovering Older Versions of a File
+
+Jennifer has made changes to the Python script that she has been working on for weeks, and the
+modifications she made this morning "broke" the script and it no longer runs. She has spent
+\~ 1hr trying to fix it, with no luck...
+
+Luckily, she has been keeping track of her project's versions using Git! Which commands below will
+let her recover the last committed version of her Python script called
+`data_cruncher.py`?
+
+1. `$ git checkout HEAD`
+
+2. `$ git checkout HEAD data_cruncher.py`
+
+3. `$ git checkout HEAD~1 data_cruncher.py`
+
+4. `$ git checkout <unique ID of last commit> data_cruncher.py`
+
+5. Both 2 and 4
+
+:::::::::::::::  solution
+
+## Solution
+
+The answer is (5)-Both 2 and 4.
+
+The `checkout` command restores files from the repository, overwriting the files in your working
+directory. Answers 2 and 4 both restore the *latest* version *in the repository* of the file
+`data_cruncher.py`. Answer 2 uses `HEAD` to indicate the *latest*, whereas answer 4 uses the
+unique ID of the last commit, which is what `HEAD` means.
+
+Answer 3 gets the version of `data_cruncher.py` from the commit *before* `HEAD`, which is NOT
+what we wanted.
+
+Answer 1 can be dangerous! Without a filename, `git checkout` will restore **all files**
+in the current directory (and all directories below it) to their state at the commit specified.
+This command will restore `data_cruncher.py` to the latest commit version, but it will also
+restore *any other files that are changed* to that version, erasing any changes you may
+have made to those files!
+As discussed above, you are left in a *detached* `HEAD` state, and you don't want to be there.
+
+
+
+:::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::  challenge
+
+## Reverting a Commit
```

Comment:
This is displayed as a knowledge check with a solution, but it is presenting new information about `git revert` that isn't shown above this. This is makes how these lessons are structured harder to follow (e.g., are there callout boxes that I can safely skip?). 

This is a bigger question about structure that might be good to talk about and, if it makes sense, apply to all lessons.

Reply:
It's displayed on mine as a challenge. I'm actually going to cut this section since we'll be putting it in the confidentiality section later on anyways.